trieste/models/gpflux/interface.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+import tensorflow as tf
+from gpflow.base import Module
+
+from ...types import TensorType
+from ..interfaces import ProbabilisticModel
+
+
+class GPfluxPredictor(ProbabilisticModel, tf.Module, ABC):
+    """A trainable wrapper for a GPflux deep Gaussian process model. The code assumes subclasses
+    will use the Keras `fit` method for training, and so they should provide access to both a
+    `model_keras` and `model_gpflux`. Note: due to Keras integration, the user should remember to
+    use `tf.keras.backend.set_floatx()` with the desired value (consistent with GPflow) to avoid
+    dtype errors."""
+
+    @property
+    @abstractmethod
+    def model_gpflux(self) -> Module:
+        """The underlying GPflux model."""
+
+    @property
+    @abstractmethod
+    def model_keras(self) -> tf.keras.Model:
+        """Returns the compiled Keras model for training."""
```

Comment:
I personally am in favor of keeping it public in case the user wants to use the trained models afterwards for something, such as plotting or predicting, although probably the keras model can be private since that's used only for training.

Reply:
That might be nice. The flipside though is that our internal Trieste code may become dependant on some quirky gpflux model function. GPflux is relatively new and you may expect its API to change in non-compatible way, and it will be difficult to update Trieste if that happens. Ensuring model is private behind the wrapper shall prevent such issues.

Trieste requires user to define the model themselves anyway, and pass it to `optimize`. So if they want the model instance later, they can just hold on it to, right?

Reply:
I've changed the keras model to be private for now

Reply:
how did you change it? or the code isnt updated yet...